sql/core/src/main/scala/org/apache/spark/sql/execution/QueryExecution.scala

```diff
@@ -66,30 +66,24 @@ class QueryExecution(
   lazy val withCachedData: LogicalPlan = {
```

Comment:
Maybe not necessary, but should we clone `logical` too before sending to analyzer?

Reply:
yea I think we should